### PR TITLE
Add pdf sync and invalidate old cloudflare cache

### PR DESCRIPTION
**Pull request by Matteo-it**
## What This PR Changes

Actually we can't replace a datasheet file and keep the same url, as the files are distributed on a CDN. Since we decided to use static file for datasheet, we need to implement an action to invalidate old cache without store it.

### CODE REVIEW

**Comment by Matteo-it** [comment]
@Serjlee you have access to repo setting page to add token and secrets?

File: .github/workflows/deploy-production.yml [hunk 1]
```diff
@@ -75,6 +75,15 @@ jobs:
       - name: Sync all cacheable assets
         run: aws s3 sync --cache-control "public, max-age=31536000, immutable" --include "*.css" --include="*.js" --include="*.gif" --include="*.png" --include="*.svg" --exclude "*.html" --exclude="sw.js" --exclude="*.json" --delete public/ s3://${{ secrets.PRODUCTION_BUCKET_NAME }}/
 
-      - name: Sync all non-cacheable assets
```
Comment (Serjlee) [inline .github/workflows/deploy-production.yml:78]: Typo? I'd restore the old name

File: .github/workflows/deploy-production.yml [hunk 2]
```diff
@@ -75,6 +75,15 @@ jobs:
...
         run: aws s3 sync --cache-control "public, max-age=0, must-revalidate" --include "*.html" --include="sw.js" --include="*.json" --include "*.css" --exclude="*.js" --exclude="*.gif" --exclude="*.png" --exclude="*.svg" --delete public/ s3://${{ secrets.PRODUCTION_BUCKET_NAME }}/
+
+      - name: Sync PDF
+        run: aws s3 sync --cache-control "public, max-age=86400, must-revalidate" --include "*.pdf" --exclude="*.js" --exclude="*.gif" --exclude="*.png" --exclude="*.svg" --exclude="*.css" --exclude="*.html" --exclude="*.json" --exclude="sw.json" --delete public/ s3://${{ secrets.PRODUCTION_BUCKET_NAME }}/
+
+      - name: Invalidate cloudflare cache
+        uses: nathanvaughn/actions-cloudflare-purge@master
+        with:
+          cf_zone: ${{ secrets.CLOUDFLARE_ZONE }}
+          cf_auth: ${{ secrets.CLOUDFLARE_AUTH_KEY }}
```
Comment (Serjlee) [inline .github/workflows/deploy-production.yml:89]:
```suggestion
          cf_auth: ${{ secrets.CLOUDFLARE_PURGE_API_TOKEN }}
```

File: .github/workflows/deploy-staging.yml [hunk 3]
```diff
@@ -73,8 +73,17 @@ jobs:
...
+        run: aws s3 sync --cache-control "public, max-age=0, must-revalidate" --include "*.html" --include="sw.js" --include="*.json" --include "*.css" --exclude="*.js" --exclude="*.gif" --exclude="*.png" --exclude="*.svg" --exclude="*.pdf" --delete public/ s3://${{ secrets.STAGING_BUCKET_NAME }}/
+
+      - name: Sync PDF
+        run: aws s3 sync --cache-control "public, max-age=86400, must-revalidate" --include "*.pdf" --exclude="*.js" --exclude="*.gif" --exclude="*.png" --exclude="*.svg" --exclude="*.css" --exclude="*.html" --exclude="*.json" --exclude="sw.json" --delete public/ s3://${{ secrets.STAGING_BUCKET_NAME }}/
+
+      - name: Invalidate cloudflare cache
+        uses: nathanvaughn/actions-cloudflare-purge@master
+        with:
+          cf_zone: ${{ secrets.CLOUDFLARE_ZONE }}
+          cf_auth: ${{ secrets.CLOUDFLARE_AUTH_KEY }}
```
Comment (Serjlee) [inline .github/workflows/deploy-staging.yml:89]:
```suggestion
          cf_auth: ${{ secrets.CLOUDFLARE_PURGE_API_TOKEN }}
```

**Review by Serjlee — CHANGES_REQUESTED** [review]
Let's make sure to add `.pdf` as an `--exclude` of all syncs other than the pdf one, so they don't get deleted and then-reuploaded each time

File: .github/workflows/deploy-production.yml (hunk 1 above)
Comment (Matteo-it) [reply .github/workflows/deploy-production.yml:78]: Yes typo error, sorry

**Comment by Matteo-it** [comment]
> Let's make sure to add `.pdf` as an `--exclude` of all syncs other than the pdf one, so they don't get deleted and then-reuploaded each time

Updated both staging and prod actions

**Comment by Matteo-it** [comment]
Hello, @Serjlee, i've try to put this branch in dev https://github.com/arduino/docs-content/actions/runs/10106670041 but it fail. Can you help me to try to understand the problem?

**Pull request merged**